Orange/data/table.py

```diff
@@ -942,6 +966,65 @@ def collect(attr):
             conc.attributes.update(table.attributes)
         return conc
 
+    @classmethod
+    def _concatenate_horizontal(cls, tables):
+        """
+        """
+        if not tables:
+            raise ValueError('need at least one table to join')
+
+        def all_of(objs, names):
+            return (tuple(getattr(obj, name) for obj in objs)
+                    for name in names)
+
+        def stack(arrs):
+            non_empty = tuple(arr if arr.ndim == 2 else arr[:, np.newaxis]
+                              for arr in arrs
+                              if arr is not None and arr.size > 0)
+            return np.hstack(non_empty) if non_empty else None
+
+        doms, Ws, table_attrss = all_of(tables, ("domain", "W", "attributes"))
+        Xs, Ys, Ms = map(stack, all_of(tables, ("X", "Y", "metas")))
+        # pylint: disable=undefined-loop-variable
+        for W in Ws:
+            if W.size:
+                break
+
+        parts = all_of(doms, ("attributes", "class_vars", "metas"))
+        domain = Domain(*(tuple(chain(*lst)) for lst in parts))
+        table = cls.from_numpy(domain, Xs, Ys, Ms, W, ids=tables[0].ids)
+        for ta in reversed(table_attrss):
+            table.attributes.update(ta)
+
```

Comment:
The name of the new table should also be set, otherwise it is 'untitled'.
The same approach can be used as for axis=0:
```
names = [table.name for table in tables if table.name != "untitled"]
if names:
    table.name = names[0]
```

Reply:
I extracted the common code from `_concatenate_vertical` and `_concatenate_horizontal` to `concatenate`.

All changes are in the second commit.